google/cloud/storage/blob.py

```diff
@@ -2979,18 +2979,36 @@ def compose(
             >>> composed_blob.compose(blobs, if_source_generation_match=if_source_generation_match)
         """
         sources_len = len(sources)
-
         client = self._require_client(client)
         query_params = {}
 
-        if self.user_project is not None:
-            query_params["userProject"] = self.user_project
+        if isinstance(if_generation_match, list):
+            warnings.warn(
+                "if_generation_match: type list is deprecated and supported for backwards-compatability reasons only."
+                "Use if_source_generation_match instead to match source objects generations.",
+                PendingDeprecationWarning,
```

Comment:
This can be a regular `DeprecationWarning` -- we are already deprecating the usage, not "fixin' to commence" to deprecated it.  Also, we want to use `stacklevel=2` to highlight the call site.

Reply:
Done. Also revised the warning message based on #464. Thanks for catching this.